pauthabielf64/pauthabielf64.rst

```diff
@@ -542,243 +586,191 @@ by the run-time environment. There are a number of possible encoding
 schemas, each with its own trade-off. We are seeking feedback from
 platform owners about the most convenient form.
 
-Options for encoding the signing schema
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Encoding the signing schema
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-To create a signed pointer the run-time system needs to know the
-signing schema to use for the pointer. The object producer and static
-linker will need to communicate this via metadata; including at least:
+In the descriptions below the ``place`` is the operation ``P`` in
+AAELF64_ relocation descriptions. It is derived from the r_offset
+field of the relocation.
 
-* The Key, one of ``IA``, ``IB``, ``DA``, ``DB``. The ``GA`` key for
-  signing of generic data is not exposed in this ABI..
+The top 32-bits of the contents of the place is used to encode the
+signing schema for both static and dynamic relocations. This permits
+platforms using relocation compression or SHT_REL dynamic relocations
+to encode relocation addends in the bottom 32-bits. Given that the
+maximum size of link-units using the small code-model is 4 gigabytes
+this should be sufficient.
 
-* The constant discriminator value.
+.. class:: pauthabiencoding
 
-* Whether to combine address diversity with the discriminator.
++------+-------------------+----------+----------+----------+---------------+---------------------+
+| 63   | 62                | 61       | 60:59    | 58:48    |  47:32        | 31:0                |
++======+===================+==========+==========+==========+===============+=====================+
+| auth | address diversity | reserved | key      | reserved | discriminator | reserved for addend |
++------+-------------------+----------+----------+----------+---------------+---------------------+
 
-In ELF we have the following places where we can encode this
-information via a combination of.
+* ``auth`` is a single bit to denote that the value represents metadata
 
-* The relocation code.
+  * Review Question: Do we need this? It should only be present when
+    there is an AUTH variant dynamic relocation.
```

Comment:
I don't think we need it. I never had a need for it in my prototype.

Reply:
Thanks, I'll update and remove it.